hnn_core/network.py

```diff
@@ -288,6 +288,10 @@ class Network(object):
     legacy_mode : bool
         Set to True by default to enable matching HNN GUI output when drives
         are added suitably. Will be deprecated in a future release.
+    mesh_shape : tuple of int
```

Comment:
looks good. Could we update the tests now? So we don't use `net._paras['N_pyr_x'] = 3` anywhere ...

Reply:
I've made this change! But it has made me appreciate how deeply woven the `params` dict is in our tests sadly...